gfx_scene.h

```diff
@@ -1804,13 +1805,18 @@ class GfxSceneInternal
                 for(size_t i = 0; i < gltf_node->children_count; ++i)
                 {
                     std::map<cgltf_node const *, uint64_t>::const_iterator const it2 = animated_nodes.find(gltf_node->children[i]);
-                    if(it2 != animated_nodes.end()) children.push_back((*it2).second);
+                    GFX_ASSERT(it2 != animated_nodes.end());
+                    children.push_back((*it2).second);
                 }
                 if(animated_node != nullptr)
                 {
-                    animated_node->scale_ = S;
-                    animated_node->rotate_ = R;
-                    animated_node->translate_ = T;
+                    animated_node->translate_ = local_transform[3];
+                    for (int i = 0; i < 3; i++)
+                        animated_node->scale_[i] = glm::length(glm::dvec3(local_transform[i]));
+                    const glm::dmat3 rotMtx(glm::dvec3(local_transform[0]) / animated_node->scale_[0],
+                        glm::dvec3(local_transform[1]) / animated_node->scale_[1],
+                        glm::dvec3(local_transform[2]) / animated_node->scale_[2]);
+                    animated_node->rotate_ = glm::quat_cast(rotMtx);
```

Comment:
Nodes that had a fixed transform as well as an animated transform where nor correctly being set. This adds the fixed transform parts.

Reply:
Great fix, thanks.